cogs/fun.py

```
@@ -0,0 +1,83 @@
+                async with session.get('https://www.reddit.com/r/memes/random.json') as r:
+                    data = await r.json()
+                    await i.response.send_message(data[0]['data']['children'][0]['data']['url'])
+        except Exception as e:
+            await i.response.send_message(f'{e}')
+
+    
+
+    @nextcord.slash_command(name="joke", description="Posts a random joke from the joke api")
+    async def joke(self, i: Interaction):
+        async with aiohttp.ClientSession() as session:
+            async with session.get('https://official-joke-api.appspot.com/jokes/random') as r:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)